apps/backend/src/users/users.service.ts

```diff
@@ -1,17 +1,55 @@
-import { Injectable } from '@nestjs/common';
+import {
+  BadRequestException,
+  UnauthorizedException,
+  Injectable,
+} from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { Repository } from 'typeorm';
+import { MongoRepository } from 'typeorm';
 
 import { User } from './user.entity';
+import { Status } from './types';
+import { ObjectId } from 'mongodb';
 
 @Injectable()
 export class UsersService {
   constructor(
     @InjectRepository(User)
-    private usersRepository: Repository<User>,
+    private usersRepository: MongoRepository<User>,
```

Comment:
:question: Is there a reason for preferring `MongoRepository` over `Repository`?

Reply:
As far as I understand, `Repository` has some bugs with MongoDB operators. For example, I ran into an issue querying with the OR operator: `usersRepository.find({ where: [{ field1: value1 }, { field2: value2 }] })` kept failing. `MongoRepository` is designed to use native MongoDB $ operators. I was able to run the above with no issue using `usersRepository.find({ where: { $or: [{ field1: { $eq: value1 } }, { field2: { $eq: value2 } }] } })`. We could keep using `Repository`, but I think it could get glitchy.